### PR TITLE
Fix#362 : Fixed order of GHAs

### PR DESCRIPTION
**Pull request by RyanWalker277**
Related to #362 .

Hey @tushar5526 @ziyaointl ! I have restructured the build tests and compiled all of them into one singe `GHA`.Can you test this?

Explaination of changes: added a `needs` parameter in every job connecting it with the previous job.

### CODE REVIEW

File: .github/workflows/build-tests.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest
+          sudo apt-get install python-opengl libglfw3 -y
```
Comment (tushar5526) [inline .github/workflows/build-tests.yml:49]: This has to be installed first. See the existing GHAs and keep them the same order. That's why the actions are failing as well.

**Review by tushar5526 — CHANGES_REQUESTED** [review]
GHAs are failing. Install the apt dependencies before hand.

File: .github/workflows/build-tests.yml (hunk 1 above)
Comment (RyanWalker277) [reply .github/workflows/build-tests.yml:49]: The order is the same as existing GHAs.The `python-opengl` isn't installing in the ubuntu image for some reason.I even tried installing the package through a seperate GHA on another repo yet it failed , seems like an issue with the apt-package itself.Can you re-run the pytest action on the latest commit and tell if the same issue persists there as well?

Comment (tushar5526) [reply .github/workflows/build-tests.yml:49]: Hey, I just checked can you change it to `python3-opengl` from `python-opengl` and see if it works? I think the name has changed in newer version. https://packages.ubuntu.com/jammy/python/python3-opengl

Comment (RyanWalker277) [reply .github/workflows/build-tests.yml]: Ya that was the issue.Now it's working fine.

**Review by tushar5526 — APPROVED** [review]

**Comment by tushar5526** [comment]
Thank you for the contribution

**Pull request merged**